freqtrade/freqtradebot.py

```diff
@@ -119,16 +135,27 @@ def worker(self, old_state: State = None) -> State:
             if state == State.RUNNING:
                 self.rpc.startup_messages(self.config, self.pairlists)
 
+        throttle_secs = self.config.get('internals', {}).get(
+            'process_throttle_secs',
+            constants.PROCESS_THROTTLE_SECS
+        )
+
         if state == State.STOPPED:
-            time.sleep(1)
+            # Ping systemd watchdog before sleeping in the stopped state
+            if self._sd_notify:
+                logger.debug("sd_notify: WATCHDOG=1\\nSTATUS=State: STOPPED.")
+                self._sd_notify.notify('WATCHDOG=1\nSTATUS=State: STOPPED.')
```

Comment:
I think you should use the same type of quotes for both messages ... 
now the same message needs different escaping.

Same ~10 lines below

Reply:
I've got a custom for python -- I use double quotes for strings that can be internationalized/localized (i.e. different user messages) and single quotes for non-internationalizable strings (diff. API messages, states, string constants etc.) So the debug messages went into double quotes "automatically" here...

* Made them (quotes) same
* Added a message to systemd for reconfiguration. (should be issued from telegram, no handling of SIGHUP needed for systemd or similar functionality to initiate reconfiguration from cli)
* I decided to not add handling of the SIGHUP for starting reconfiguration from cli in order to keep the code simple -- it does not make much sense since reconfiguration anyway looks mostly like stop/start (cleanup and creation of a new freqtradebot)